cmd/update-main.go

```diff
@@ -274,8 +277,8 @@ func mainUpdate(ctx *cli.Context) {
 		os.Exit(-1)
 	}
 
-	if older != time.Duration(0) {
-		log.Println(colorizeUpdateMessage(downloadURL, older))
+	if updateMsg := computeUpdateMessage(downloadURL, older); updateMsg != "" {
```

Comment:
The `older` value is being used in the `computeUpdateMessage` to return an appropriate `updateMsg`:

```
	if downloadURL == "" || older <= 0 {
		return ""
	}
```

(would it help if condition were written as `older <= 0 || downloadURL == ""`? I don't think its a big difference - we are checking downloadURL == "" just as an additional (sanity) check).

 If we check `older` outside here again, and then conditionally call `computeUpdateMessage` you are just introducing another 0 value in a different way that you already said could be avoided. I think this is fine.